### PR TITLE
[MM-43649] - Ability for end users to notify admin to upgrade workspace

### PR DESCRIPTION
**Pull request by AGMETEOR**
#### Summary
Adds the ability to notify admins to upgrade from the pricing modal

#### Ticket Link
[MM-43649](https://mattermost.atlassian.net/browse/MM-43649)

#### Related Pull Requests

- Has server changes (https://github.com/mattermost/mattermost-server/pull/20338)

#### Screenshots
Screenshot 2022-06-02 at 19 46 07

Screenshot 2022-06-02 at 19 44 45

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by harshilsharma63** [comment]
Screenshot 2022-06-03 at 11 20 40 AM

@AGMETEOR the styling is a bit off because we're applying different width borders on the same element. We need to create a wrapper-div just for that special left border to avoid the skewed edge.

**Comment by AGMETEOR** [comment]
> Screenshot 2022-06-03 at 11 20 40 AM
> 
> @AGMETEOR the styling is a bit off because we're applying different width borders on the same element. We need to create a wrapper-div just for that special left border to avoid the skewed edge.

@harshilsharma63 fixed the styling
Screenshot 2022-06-06 at 11 55 34

**Review by marianunez — COMMENTED** [review]
Thanks @AGMETEOR! Overall this LGTM, will complete the review when the requested changes on the server are done

**Review by neallred — APPROVED** [review]
LGTM, thanks @AGMETEOR !

File: components/root/root.jsx
Comment (AGMETEOR) [reply components/root/root.jsx:355]: `custom_upgrade_notification` will not work because that field has a max char length of 21 in the DB

File: components/custom_open_pricing_modal_post_renderer/index.tsx
Comment (AGMETEOR) [reply components/custom_open_pricing_modal_post_renderer/index.tsx:10]: We can move it to the pricing modal folder, but this is intended to be used by the other products (playbooks and boards) as well

Comment (marianunez) [reply components/custom_open_pricing_modal_post_renderer/index.tsx:10]: Makes sense, can we just shorten the name as suggested above?

File: components/root/root.jsx
Comment (marianunez) [reply components/root/root.jsx:344]: I see, then I suggest we drop the `custom` part? ` up_notification` seems a bit confusion 

```suggestion
        this.props.actions.registerCustomPostRenderer('upgrade_notification', CustomOpenPricingModalPostRenderer, 'upgrade_post_message_renderer');
```

File: e2e/cypress/tests/integration/enterprise/cloud/billing/pricing_modal_spec.js
Comment (marianunez) [inline e2e/cypress/tests/integration/enterprise/cloud/billing/pricing_modal_spec.js:57]: What tests could we add to test the new post notification?

File: components/root/root.jsx
Comment (AGMETEOR) [reply components/root/root.jsx:344]: Can't drop it because the server expects it to have that prefix. See `model.PostCustomTypePrefix`

File: e2e/cypress/tests/integration/enterprise/cloud/billing/pricing_modal_spec.js
Comment (AGMETEOR) [reply e2e/cypress/tests/integration/enterprise/cloud/billing/pricing_modal_spec.js:51]: I have added cypress test to send the ping and also check admin DMs for the system-bot ping

Comment (marianunez) [reply e2e/cypress/tests/integration/enterprise/cloud/billing/pricing_modal_spec.js:51]: @AGMETEOR I logged in as a regular user and I don't see the upgrade button on the test server of this PR. @annelieseklein is the expectation that the Upgrade button is available for end users? It doesn't seem like it in the designs. 

The ability to see the pricing modal and ask for upgrade will be from the link @pvev will be adding on the main menu.

Comment (AGMETEOR) [reply e2e/cypress/tests/integration/enterprise/cloud/billing/pricing_modal_spec.js:51]: For this PR, the Upgrade. button shows

Comment (marianunez) [reply e2e/cypress/tests/integration/enterprise/cloud/billing/pricing_modal_spec.js:51]: To not show the Upgrade button for end users - I believe is the correct behavior. However, not sure how the tests below are passing though 🤔

Comment (AGMETEOR) [reply e2e/cypress/tests/integration/enterprise/cloud/billing/pricing_modal_spec.js:51]: I see what you mean. Let me try recreating this spinwick

Comment (AGMETEOR) [reply e2e/cypress/tests/integration/enterprise/cloud/billing/pricing_modal_spec.js:51]: I think something is up with the spinwick. The logic to show upgrade button nolonger requires an admin

Comment (marianunez) [reply e2e/cypress/tests/integration/enterprise/cloud/billing/pricing_modal_spec.js:51]: Was that changed in this PR?

Comment (AGMETEOR) [reply e2e/cypress/tests/integration/enterprise/cloud/billing/pricing_modal_spec.js:51]: On second thought, it could be because the subscription info is still limited to only admin. Let me check

Comment (AGMETEOR) [reply e2e/cypress/tests/integration/enterprise/cloud/billing/pricing_modal_spec.js:51]: This https://github.com/mattermost/mattermost-server/pull/20540 and https://github.com/mattermost/mattermost-server/pull/20534 will fix once they're merged

File: e2e/cypress/tests/integration/enterprise/cloud/billing/pricing_modal_spec.js
Comment (AGMETEOR) [inline e2e/cypress/tests/integration/enterprise/cloud/billing/pricing_modal_spec.js:72]: @marianunez temporarily commented out these tests until we have a way to open the pricing modal for non-admin users

File: components/global_header/right_controls/cloud_upgrade_button/index.tsx
Comment (johndavidlugtu) [reply components/global_header/right_controls/cloud_upgrade_button/index.tsx]: @AGMETEOR  @marianunez  Just wanted to be clear here -- is the question whether the upgrade button should exist and be clickable for non-admins?

Comment (AGMETEOR) [reply components/global_header/right_controls/cloud_upgrade_button/index.tsx]: Yes, that's the question.

Comment (johndavidlugtu) [reply components/global_header/right_controls/cloud_upgrade_button/index.tsx]: Yes, only for admins. The button is visible but greyed out @AGMETEOR

Comment (AGMETEOR) [reply components/global_header/right_controls/cloud_upgrade_button/index.tsx]: @johndavidlugtu by greyed out does it mean it should be disabled?

**Review by marianunez — APPROVED** [review]

File: components/global_header/right_controls/cloud_upgrade_button/index.tsx
Comment (johndavidlugtu) [reply components/global_header/right_controls/cloud_upgrade_button/index.tsx]: @AGMETEOR it looks like there was a bit of confusion. Please refer to this conversation in MM: https://community.mattermost.com/private-core/pl/ytj3h7c7dprr8mdoogom5345wo

**Comment by johndavidlugtu** [comment]
Screen Shot 2022-06-29 at 6 44 00 AM

So in this [Figma](https://www.figma.com/file/Aob5ffHtfjKzQoebR93ycA/Cloud-Freemium?node-id=2543%3A228056) design the "Notify admin" exists in both. Not sure if the screenshot above is up-to-date @AGMETEOR ?

**Comment by johndavidlugtu** [comment]
Sounds good, I've notified Pablo about this. Approving the PR now

**Pull request merged**
